src/client/datascience/jupyter/kernelVariables.ts

```diff
@@ -481,6 +497,10 @@ export class KernelVariables implements IJupyterVariables {
             if (DataViewableTypes.has(result.type)) {
                 result.supportsDataExplorer = true;
             }
+
+            if (SliceableTypes.has(result.type)) {
+                result.supportsSlicing = true;
```

Comment:
I don't think this needs to be an attribute of the variable, this is something that can be derived.
Also if this had to be an attirbute, then it should come all the way from the backend (Python) not something we just tack onto the result (else feels hackish).

Reply:
This was modeled on `supportsDataExplorer` but in retrospect can be done in the UI. But I don't really agree with the idea of this attribute coming from the backend. The UI can just look at the `type` property for this to figure out whether to show the slice controls.

Reply:
> he UI can just look at the type property for this to figure out whether to show the slice controls.

Agred. I only suggested sending the stuff from the backend, if you still wanted to keep the property on the object. 
So, my understanding is that this property will go away and we'll have a simple function call to determien whether a variable supports slicing or not.
Is that correct?

Reply:
Yes that's right.

Reply:
Oh sorry, I just noticed. Currently we do not send the `type` property on `IDataFrameInfo` because it's just not necessary, so we must add either a `supportsSlicing` or `type` property. Assuming you're fine with that (we'll still be adding a property), I think that's still better because slicing is a UI operation and the backend shouldn't need to know about that if the UI can figure it out.

Reply:
I think we should send the `type` property, and let other information/behaviours be derived from that information. I.e. `supporsSlicing` can be derived from that information.

Reply:
This way its more consistent, to send type from backend, then to extension, then to UI. instead of augmenting with custom information from extension.

Reply:
One disadvantage of this approach is that the UI now knows about which specific types we need to enable slicing for. That's language-specific. Not a problem now, just one more thing to disentangle if/when we start looking at supporting other languages in the data viewer...